sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTableSupport.scala

```diff
@@ -369,9 +371,6 @@ private[parquet] class MutableRowWriteSupport extends RowWriteSupport {
       case DateType => writer.addInteger(record.getInt(index))
       case TimestampType => writeTimestamp(record.getLong(index))
       case d: DecimalType =>
-        if (d.precisionInfo == None || d.precisionInfo.get.precision > 18) {
-          sys.error(s"Unsupported datatype $d, cannot write to consumer")
-        }
```

Comment:
Overseen, bug. We can't serialize without that info, parquet requires it (and mixed scale would be complicated).

PS: do you know if there is any interest on allowing mixed Decimal in parquet?

Am 24. Juni 2015 09:36:11 MESZ, schrieb Cheng Lian <notifications@github.com>:
>> @@ -369,9 +371,6 @@ private[parquet] class MutableRowWriteSupport
>extends RowWriteSupport {
>>        case DateType => writer.addInteger(record.getInt(index))
>>        case TimestampType => writeTimestamp(record.getLong(index))
>>        case d: DecimalType =>
>> -        if (d.precisionInfo == None || d.precisionInfo.get.precision
>> 18) {
>> -          sys.error(s"Unsupported datatype $d, cannot write to
>consumer")
>> -        }
>
>Don't we need to consider the case where `d.precisionInfo == None` now?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/apache/spark/pull/6796/files#r33123740

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.